termlink/gsea.py

```diff
@@ -0,0 +1,97 @@
+"""Handles Gene Sets conversion.
+
+This module provides methods to extract, transform and load relationships
+defined by the Geneset dataset.
+
+The download files for Geneset are provided at http://software.broadinstitute.org/gsea/msigdb/collections.jsp.
+"""
+import os
+import csv
+import json
+
+from urllib.parse import urlparse
+
+from termlink.commands import SubCommand
+from termlink.services import RelationshipService
+
+
+class Command(SubCommand):
+    """
+    A command executor for GSEA operations
+    """
+
+    @staticmethod
+    def execute(args):
+        """
+        Prints a JSON array of `Relationship` objects to stdout
+
+        Args:
+            args: `argparse` parsed arguments
+        """
+        uri = urlparse(args.uri)
+        service = Service(uri)
+        rows = service.get_relationships()
+        for row in rows:
+            print(json.dumps(row))
+
+
+def build_node(gsea, i):
+    """
+     Extracts single target/source system from a tab delimited/iterator
+    """
+    return {'type': 'coding', 'display': gsea[i], 'system': 'http://www.broadinstitute.org/gsea/msigdb',
```

Comment:
You should use the classes in `termlink.models` to enforce types. See the `termlink.hpo` module for examples.